Carbon/navigation/screens/Settings/SettingsScreen.js

```diff
@@ -1,50 +1,169 @@
 import React, { useState, useEffect } from 'react';
-import { View, Text, Button, StyleSheet } from 'react-native';
+import { View, Text, Button, StyleSheet, ScrollView, Image, Pressable } from 'react-native';
 import UsernameInput from '../../../components/UsernameInput';
 import PasswordInput from '../../../components/PasswordInput';
 import { Colors } from '../../../styling/Colors';
 import { logout } from '../../../util/LoginManager';
 import ChangeUsernameButton from '../../../components/ChangeUsernameButton';
 import ChangePasswordButton from '../../../components/ChangePasswordButton';
-import { changeUsername, changePassword } from '../../../util/UpdateAccountSettings';
+import { changeUsername, changePassword, changePFP } from '../../../util/UpdateAccountSettings';
+import { FlatList, TouchableOpacity } from 'react-native-gesture-handler';
+import pfps from '../../../assets/profile-icons/index'
+import { API_URL } from '../../../config/Api';
+import { getToken } from '../../../util/LoginManager';
+import Svg, { Defs, Rect }  from 'react-native-svg';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import LoadingIndicator from '../../../components/LoadingIndicator';
 
 const NonBreakingSpace = () => <Text>{'\u00A0'}</Text>;
 
+const USER_API = API_URL + 'user/';
+
 const SettingsScreen = ({ navigation }) => {
+
+    // User data
+    const [user, setUser] = useState(null);
+    const [loadingUser, setLoadingUser] = useState(true);
+
+    // Changing user data
     const [username, setUsername] = useState('');
     const [oldPassword, setOldPassword] = useState('');
     const [newPassword, setNewPassword] = useState('');
 
+    // Change pfp modal
+    const [modalVisible, setModalVisible] = useState(false);
+
     async function handleUsernameChange() {
         await changeUsername(username);
+        await fetchUser();
     }
     async function handlePasswordChange() {
         await changePassword(oldPassword, newPassword);
+        await fetchUser();
```

Comment:
thanks for doing this man! I was wondering if we should do something similar to update the info on our user